nat-lab/tests/utils/connection_tracker.py

```diff
@@ -92,8 +102,10 @@ def __init__(
         self._connection: Connection = connection
         self._config: Optional[List[ConnectionTrackerConfig]] = configuration
         self._events: List[FiveTuple] = []
+        self._lock: Lock = threading.Lock()
```

Comment:
Do we need this lock here? Python asyncio default runtime on a single threaded event loop, unlike rust's multi thread pool.

Reply:
Would you suggest using `asyncio.Lock()` instead then or just remove it all together?
my intention was to make sure we wait for all of the stdout lines to be processed before checking the events in `get_out_of_limits`, with the added bonus of being thread-safe

Reply:
that or Event